public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt

```diff
@@ -176,8 +176,6 @@ data class CustomerInfo constructor(
         result = 31 * result + nonSubscriptionTransactions.hashCode()
         result = 31 * result + allExpirationDatesByProduct.hashCode()
         result = 31 * result + allPurchaseDatesByProduct.hashCode()
-        result = 31 * result + requestDate.hashCode()
```

Comment:
Looks like this was removed from `equals` for the test `Given two empty purchaser infos with different request dates, both are equal`, but not from here.

Reply:
Good catch! We did this https://github.com/RevenueCat/purchases-android/pull/385/files#diff-42bdcf7858bdb44e6b3e294fb265c075ad1d939778fb7c16fa0d60f40453f28bR141 for StoreProduct, so we prevent this sort of issue from happening and we use the automatic equals and hashcode that data classes provide. I think this is a good opportunity to do the same for CustomerInfo

Reply:
But in this case we don't want to compare `requestDate`? At least that's what that test is checking. So we don't want the default implementation, right?

Reply:
Exactly. So for that we can have all fields but `requestDate` to be part of the `ComparableData` constructor. Then override the default's implementation of `CustomerInfo`'s `equals` like `override fun equals(other: Any?) = other is CustomerInfo && ComparableData(this) == ComparableData(other)`. That way we leverage the default's implementation of data classes via `ComparableData` and it's more clear that `requestDate` shouldn't be used for comparing.

Reply:
Oooh I see what you were saying. Changed!